### DISCUSSION TITLE
re-running a playbook

### DISCUSSION THREAD

**Discussion by MagnaFrisia90** [HCL-TECH-SOFTWARE/connections-automation#341] [Q&A] [answered]
Hello,
I have just tried to setup Connections "all-in-one".
Ansible ran into some errors e.g. wrong filename for TDI-installer. So I corrected the filename & just re-run the ansible-playbook.
It seems like this is not supported: the next run it skips some important parts like "TASK [connections-wizards : Create installation folder]" which result in the following error : 

> TASK [connections-wizards : Run database command /opt/IBM/db2/V11.5/bin/db2 -td@ -vf /opt/HCL/DB2Wizards/Wizards/connections.sql/activities/db2/runstats.sql] ***
> fatal: [solcon31.solvito.cloud]: FAILED! => {"changed": true, "cmd": "/opt/IBM/db2/V11.5/bin/db2 -td@ -vf /opt/HCL/DB2Wizards/Wizards/connections.sql/activities/db2/runstats.sql", "delta": "0:00:01.023996", "end": "2024-08-08 15:28:49.547927", "failed_when_result": true, "msg": "non-zero return code", "rc": 8, "start": "2024-08-08 15:28:48.523931", "stderr": "", "stderr_lines": [], "stdout": "DB21005E  An error occurred while accessing the file \n\"/opt/HCL/DB2Wizards/Wizards/connections.sql/activities/db2/runstats.sq\".", "stdout_lines": ["DB21005E  An error occurred while accessing the file ", "\"/opt/HCL/DB2Wizards/Wizards/connections.sql/activities/db2/runstats.sq\"."]}

I am now wondering:
Is re-running playbooks supported ? - if yes:
a) Why did the playbook remove the folder "/opt/HCL/DB2Wizards/Wizards/" after the failed run ?
b) Why does the playbook wants to re-run the "runstats.sql" once more ?
c) I found that setting "cnx_force_repopulation=true" 'fixes' this problem. But it will re-create all databases which costs much time. Especially when you are in a PoC/trial&error phase this is really annoying. Can I somehow tell the installer to keep the databases and just re-download the Wizards to just being able to continue with the installation ?
[update to c)] I was able to adjust vars.yml & tasks.yml to skip the database creation by adding a custom-boolean-variable and add it to the "when"-part inside the yml.

Thanks in advance.

Kind regards

**Comment by sabrina-yee** [comment, marked as answer]
Hi @MagnaFrisia90, yes, re-running the database steps (i.e.`setup-connections-wizards.yml` within the `setup-connections-complete.yml` playbook) is supported.  The fact that it fails in your scenario is a bug, as it should account for running those `runstats.sql` scripts again.  We'll work on a fix.  
The reason it runs the `runstats.sql` scripts more than once is to accommodate the use case described in this topic (https://opensource.hcltechsw.com/connections-doc/v8-cr7/admin/admin/t_admin_db_maintain_stats.html).  If the databases have been successfully created but the `setup-connections-complete.yml` playbook needs to be rerun, you can set `setup_connections_wizards: false` in `all.yml` to bypass the database steps altogether.
